yarn-project/end-to-end/src/e2e_key_registry.test.ts

```diff
@@ -181,84 +183,321 @@ describe('SharedMutablePrivateGetter', () => {
 
       const rawLogs = await pxe.getUnencryptedLogs({ txHash });
 
-      expect(Fr.fromBuffer(rawLogs.logs[0].log.data)).toEqual(new Fr(12));
+      expect(Fr.fromBuffer(rawLogs.logs[0].log.data)).toEqual(poseidon2Hash(masterNullifierPublicKey.toFields()));
     });
   });
 
-  describe('key rotation flow', () => {
-    it('we rotate the nullifier key', async () => {
-      // This changes
-      const newMasterNullifierPublicKey = new Fr(910);
+  describe('key rotation flows', () => {
+    const firstNewMasterNullifierPublicKey = Point.random();
+
+    describe('normal key rotation flow', () => {
```

Comment:
Given that all the 3 cases in this describe block are interdependent I think it would make sense to just have them as one test case.